lib/src/base_client.dart

```diff
@@ -55,16 +56,15 @@ abstract class BaseClient implements Client {
 
   Future<Response> send(Request request);
 
+  void close() {}
+
   /// Throws an error if [response] is not successful.
-  void _checkResponseSuccess(url, Response response) {
+  static void _checkResponseSuccess(url, Response response) {
     if (response.statusCode < 400) return;
-    var message = "Request to $url failed with status ${response.statusCode}";
-    if (response.reasonPhrase != null) {
-      message = "$message: ${response.reasonPhrase}";
+    var message = 'Request to $url failed with status ${response.statusCode}';
```

Comment:
Please don't change the quotes. The change isn't worth the churn, and trying to keep quotes consistent isn't worth the mental overhead. Worth noting that it's explicitly *not* part of the style guide.